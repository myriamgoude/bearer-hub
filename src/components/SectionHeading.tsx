import * as React from 'react'
import styled from '@emotion/styled'

const StyledDiv = styled.div`
  width:100%;
  flex-shrink: 0;
  margin-left: auto;
  text-align: center;
`

interface ISectionHeadingProps {
  primaryText: string,
  secondaryText?: string
}

const SectionHeading = (ISectionHeadingProps:ISectionHeadingProps) => (
  <StyledDiv>
    <h2>{ISectionHeadingProps.primaryText}</h2>
    <h3>{ISectionHeadingProps.secondaryText}</h3>
  </StyledDiv>
)

export default SectionHeading
