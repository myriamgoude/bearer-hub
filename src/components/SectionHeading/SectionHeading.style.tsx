import { css } from '@emotion/core'
import { getEmSize } from '../../styles/mixins'

export default {
  root: css`
    width: 100%;
    flex-shrink: 0;
    text-align: left;
    display: block;
    margin-bottom: ${getEmSize(24)};

    & > h2 {
      margin-top: 2rem;
    }
  `,
  heading: css`
    display: inline;
  `,
  alignLeft: css`
    text-align: left;
  `,
  alignCenter: css`
    text-align: center;
  `,
  alignRight: css`
    text-align: right;
  `
}
