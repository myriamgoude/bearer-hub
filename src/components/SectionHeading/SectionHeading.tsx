import * as React from 'react'

import Text from '../Text'

import styles from './SectionHeading.style'

interface ISectionHeadingProps {
  primaryText?: string | any
  secondaryText?: string
  className?: string
  align?: string
  tag?: string
  css?: any
  children?: any
  tooltip?: any
}

const SectionHeading = (props: ISectionHeadingProps) => {
  let alignment
  switch (props.align) {
    case 'left':
      alignment = styles.alignLeft
      break
    case 'right':
      alignment = styles.alignRight
      break
    case 'center':
      alignment = styles.alignCenter
      break
    default:
      alignment = styles.alignCenter
      break
  }
  return (
    <div css={[styles.root, alignment, props.css && props.css]} className={props.className && props.className}>
      <Text tag={props.tag ? props.tag : 'h2'}>
        {props.primaryText} {props.tooltip}
      </Text>
      <h3 css={styles.heading}>{props.secondaryText}</h3>
      {props.children}
    </div>
  )
}
export default SectionHeading
