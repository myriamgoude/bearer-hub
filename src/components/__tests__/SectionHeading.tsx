import * as React from 'react'
import renderer from 'react-test-renderer'
import SectionHeading from '../SectionHeading/SectionHeading'

describe('SectionHeading', () => {
  it('renders correctly with tooltip text', () => {
    const tree = renderer
      .create(
        <SectionHeading
          tooltip="Drink me"
          align="left"
          primaryText="Alice's Adventures"
          secondaryText="Through the Looking Glass"
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly without tooltip text', () => {
    const tree = renderer
      .create(
        <SectionHeading align="left" primaryText="Alice's Adventures" secondaryText="Through the Looking Glass" />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
