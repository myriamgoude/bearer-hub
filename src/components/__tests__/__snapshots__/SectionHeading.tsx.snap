// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHeading renders correctly with tooltip text 1`] = `
<div
  css={
    Array [
      Object {
        "map": undefined,
        "name": "3pg1pf",
        "next": undefined,
        "styles": "
    width: 100%;
    flex-shrink: 0;
    text-align: left;
    display: block;
    margin-bottom: 1.5rem;
  ",
      },
      Object {
        "map": undefined,
        "name": "139b19o",
        "next": undefined,
        "styles": "
    text-align: left;
  ",
      },
      undefined,
    ]
  }
>
  <h2
    css={
      Array [
        Object {
          "map": undefined,
          "name": "nd9hyy",
          "next": undefined,
          "styles": "
    display: inline-block;
    color: #030D36;
    font-size: 1.125rem;
  ",
        },
        Object {
          "map": undefined,
          "name": "9oecbb",
          "next": undefined,
          "styles": "
    font-size: 16px;
  ",
        },
        undefined,
        undefined,
        undefined,
        false,
        Object {
          "map": undefined,
          "name": "elpby5",
          "next": undefined,
          "styles": "
    font-size: 32px;
    color: #030D36;
    font-family: 'ProximaNova-Semibold';
  ",
        },
        false,
        false,
        false,
        undefined,
      ]
    }
  >
    Alice's Adventures
     
    Drink me
  </h2>
  <h3
    css={
      Object {
        "map": undefined,
        "name": "89fcrv",
        "next": undefined,
        "styles": "
    display: inline;
  ",
      }
    }
  >
    Through the Looking Glass
  </h3>
</div>
`;

exports[`SectionHeading renders correctly without tooltip text 1`] = `
<div
  css={
    Array [
      Object {
        "map": undefined,
        "name": "3pg1pf",
        "next": undefined,
        "styles": "
    width: 100%;
    flex-shrink: 0;
    text-align: left;
    display: block;
    margin-bottom: 1.5rem;
  ",
      },
      Object {
        "map": undefined,
        "name": "139b19o",
        "next": undefined,
        "styles": "
    text-align: left;
  ",
      },
      undefined,
    ]
  }
>
  <h2
    css={
      Array [
        Object {
          "map": undefined,
          "name": "nd9hyy",
          "next": undefined,
          "styles": "
    display: inline-block;
    color: #030D36;
    font-size: 1.125rem;
  ",
        },
        Object {
          "map": undefined,
          "name": "9oecbb",
          "next": undefined,
          "styles": "
    font-size: 16px;
  ",
        },
        undefined,
        undefined,
        undefined,
        false,
        Object {
          "map": undefined,
          "name": "elpby5",
          "next": undefined,
          "styles": "
    font-size: 32px;
    color: #030D36;
    font-family: 'ProximaNova-Semibold';
  ",
        },
        false,
        false,
        false,
        undefined,
      ]
    }
  >
    Alice's Adventures
     
  </h2>
  <h3
    css={
      Object {
        "map": undefined,
        "name": "89fcrv",
        "next": undefined,
        "styles": "
    display: inline;
  ",
      }
    }
  >
    Through the Looking Glass
  </h3>
</div>
`;
